driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.connection;
+
+import com.github.luben.zstd.Zstd;
+import com.github.luben.zstd.ZstdInputStream;
+import com.mongodb.MongoInternalException;
+import org.bson.ByteBuf;
+import org.bson.io.BsonOutput;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+public class ZstdCompressor extends Compressor {
+    @Override
+    public String getName() {
+        return "zstd";
+    }
+
+    @Override
+    public byte getId() {
+        return 3;
+    }
+
+    @Override
+    public void compress(final List<ByteBuf> source, final BsonOutput target) {
+        int uncompressedSize = getUncompressedSize(source);
+
+        try {
+            byte[] singleByteArraySource = new byte[uncompressedSize];
+            copy(source, singleByteArraySource);
+
+            byte[] out = Zstd.compress(singleByteArraySource);
```

Comment:
Can we use `Zstd#compress(byte[], byte[], int)` instead of this convenience method in order to avoid an extra array copy?